### PR TITLE
Migration to Focal.

### PR DESCRIPTION
**Pull request by mbuerki**

### CODE REVIEW

**Review by HannesSommer — COMMENTED** [review]
These changes don't look backwards compatible to me. Did you try building this on Bionic with Python2?

**Comment by mbuerki** [comment]
@HannesSommer now it builds both on bionic with python 2 and focal with python 3. no idea if it's the most beautiful solution though... open for suggestions.

File: src/matplotlibcpp.cpp [hunk 1]
```diff
@@ -94,12 +94,16 @@ struct _interpreter {
...
+    char name[] = "plotting";
+#else
+    const wchar_t name[] = L"plotting";
+#endif
     Py_SetProgramName(name);   // optional but recommended
     Py_Initialize();
 
-    PyObject* pyplotname = PyString_FromString("matplotlib.pyplot");
-    PyObject* pylabname = PyString_FromString("pylab");
+    PyObject* pyplotname = PyBytes_FromString("matplotlib.pyplot");
```
[the remark this answers is not in the document]
Comment (mbuerki) [reply src/matplotlibcpp.cpp:105]: I got this from here: https://github.com/nvdla/hw/issues/185
but other sources say it should be `PyUnicode_FromString` too, so I changed it to that.

**Comment by mbuerki** [comment]
@HannesSommer I added your suggestions. it compiles on both OSs. on bionic, I can run the `mpl_example_node` and that works fine. on focal in the docker it immediately segfaults. but that's maybe because there's no xserver? is there a way to test it on focal too?

**Comment by HannesSommer** [comment]
I think you forgot to push (no new commits, yet).
Segfault without X server is already pretty bad. Maybe first compare with behavior on Bionic by prefixing the bash command with `DISPLAY= `. Does it also segfault?
You can mount the X server's socket into the docker (~`docker run -v /tmp/.X11-unix:/tmp/.X11-unix -e DISPLAY=$DISPLAY \
       -h $HOSTNAME -v $HOME/.Xauthority:/root/.Xauthority ...`). But for that to succeed you probably need to install some X client libs as well. Maybe just install xterm and try it out first.

**Review by HannesSommer — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**